### PR TITLE
Validate basis points to not be greater than 100%

### PR DESCRIPTION
**Pull request by MiksuJak**

### CODE REVIEW

**Comment by vanruch** [comment]
I'd also add a check for `joiningFee` in Pool

File: contracts/truefi/TrueRatingAgency.sol [hunk 1]
```diff
@@ -163,6 +163,7 @@ contract TrueRatingAgency is ITrueRatingAgency, Ownable {
      * @param newLossFactor New loss factor
      */
     function setLossFactor(uint256 newLossFactor) external onlyOwner {
+        require(newLossFactor <= 10000, "TrueRatingAgency: Loss factor cannot be greater than 100%");
```
Comment (barrutko) [inline contracts/truefi/TrueRatingAgency.sol:166]: 👍

File: contracts/truefi/TrueRatingAgency.sol [hunk 2]
```diff
@@ -172,6 +173,7 @@ contract TrueRatingAgency is ITrueRatingAgency, Ownable {
      * Burn factor decides what percentage of lost TRU is burned
      */
     function setBurnFactor(uint256 newBurnFactor) external onlyOwner {
+        require(newBurnFactor <= 10000, "TrueRatingAgency: Burn factor cannot be greater than 100%");
```
Comment (barrutko) [inline contracts/truefi/TrueRatingAgency.sol:176]: 👍

File: test/truefi/TrueRatingAgency.test.ts [hunk 3]
```diff
@@ -117,6 +117,11 @@ describe('TrueRatingAgency', () => {
         await expect(rater.connect(otherWallet).setLossFactor(1234))
           .to.be.revertedWith('caller is not the owner')
       })
+
+      it('must be less than or equal 100%', async () => {
```
Comment (barrutko) [inline test/truefi/TrueRatingAgency.test.ts:121]: 👍

File: test/truefi/TrueRatingAgency.test.ts [hunk 4]
```diff
@@ -135,6 +140,11 @@ describe('TrueRatingAgency', () => {
         await expect(rater.connect(otherWallet).setBurnFactor(1234))
           .to.be.revertedWith('caller is not the owner')
       })
+
+      it('must be less than or equal 100%', async () => {
+        await expect(rater.setBurnFactor(100 * 101))
+          .to.be.revertedWith('TrueRatingAgency: Burn factor cannot be greater than 100%')
+      })
```
Comment (barrutko) [inline test/truefi/TrueRatingAgency.test.ts:147]: 👍

**Review by barrutko — COMMENTED** [review]
@vanruch is right. There is also another method, that requires this check. `setJoiningFee` in `TrueFiPool.sol`.

**Comment by MiksuJak** [comment]
Added check on joiningFee setter

**Review by barrutko — COMMENTED** [review]
Something went wrong with merge. Some code from previous dev commits are being deleted here 😁

**Comment by MiksuJak** [comment]
> Something went wrong with merge. Some code from previous dev commits are being deleted here 😁

Collisions resolved. Please check once again, if everything necessary is being added.

**Review by vanruch — APPROVED** [review]

**Pull request merged**